project/src/com/google/daggerquery/protobuf/binding.proto

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+ */
+
+syntax = "proto2";
+package dagerquery.models;
+
+option java_package = "com.google.daggerquery.protobuf";
+option java_outer_classname = "BindingProto";
+
+message Binding {
+
+  enum Kind {
+    INJECTION = 0;
+    PROVISION = 1;
+    COMPONENT = 2;
+    COMPONENT_PROVISION = 3;
+    COMPONENT_DEPENDENCY = 4;
+    MEMBERS_INJECTOR = 5;
+    SUBCOMPONENT_CREATOR = 6;
+    BOUND_INSTANCE = 7;
+    PRODUCTION = 8;
+    COMPONENT_PRODUCTION = 9;
+    MULTIBOUND_SET = 10;
+    MULTIBOUND_MAP = 11;
+    OPTIONAL = 12;
+    DELEGATE = 13;
+    MEMBERS_INJECTION = 14;
+  }
+
+  required string component_name = 1;
+  required bool is_component     = 2;
+  required bool is_missing       = 3;
```

Comment:
Two fields `is_component` and `is_missing` needed for distinguishing three different classes: [BindingNode](https://github.com/google/dagger/blob/4a3115780c837f0857e9a14b09ee76a932d8dd80/java/dagger/internal/codegen/binding/BindingNode.java#L45), [MissingBinding](https://github.com/google/dagger/blob/4a3115780c837f0857e9a14b09ee76a932d8dd80/java/dagger/model/BindingGraph.java#L406) and [ComponentNode](https://github.com/google/dagger/blob/4a3115780c837f0857e9a14b09ee76a932d8dd80/java/dagger/model/BindingGraph.java#L443).